code/modules/food_and_drinks/food/snacks_pie.dm

```diff
@@ -44,7 +44,7 @@
 	if(ishuman(hit_atom))
 		var/mob/living/carbon/human/H = hit_atom
 		var/mutable_appearance/creamoverlay = mutable_appearance('icons/effects/creampie.dmi')
-		if(H.dna.species.limbs_id == "lizard")
+		if(/obj/item/bodypart/head/lizard in H.bodyparts)
```

Comment:
do istype

Reply:
Thats more expensive

Reply:
It's not
istype(H.head, type) isn't more expensive than searching a list of objects

Reply:
the "head" var is the mob's hat, not their head.

Reply:
This code doesn't work. This is checking if the reference /obj/item/bodypart/head/lizard is in the list, not if the list contains an object of type /obj/item/bodypart/head/lizard. The best solution here would be to have some kind of assoc list that allows you to access the bodypart at a location and use that instead with an istype.

Reply:
Fixed in commit.